foreman.te

```diff
@@ -171,190 +188,156 @@ require{
-# or to connect to remote Cockpit instance (via Remote Execution)
-allow passenger_t websm_port_t:tcp_socket name_connect;
+# It is not possible to do transition into SCL application
+# without calling a shell: https://github.com/sclorg/scl-utils/issues/31
+# These rules are for: /usr/libexec/foreman/tfm-ruby-selinux
+corecmd_exec_shell(foreman_rails_t)
+corecmd_exec_ls(foreman_rails_t)
+# The following is needed for "scl enable" which creates temporary
+# shell script /var/tmp/xxxxxxx and a subprocess.
+files_manage_generic_tmp_files(foreman_rails_t)
 
-# Allow Foreman to connect to Foreman Proxy on a defined port
-allow passenger_t foreman_proxy_port_t:tcp_socket name_connect;
+# General files read and write
+admin_pattern(foreman_rails_t, foreman_lib_t, foreman_lib_t)
+admin_pattern(foreman_rails_t, foreman_var_run_t, foreman_var_run_t)
+admin_pattern(foreman_rails_t, passenger_tmp_t, passenger_tmp_t)
 
-# Allow Foreman to connect to PostgreSQL
-corenet_tcp_connect_postgresql_port(passenger_t)
-optional_policy(`
-    postgresql_stream_connect(passenger_t)
-')
+# Gettext support
+miscfiles_read_localization(foreman_rails_t)
 
-# Allow Foreman to connect to Redis
-corenet_tcp_connect_redis_port(passenger_t)
+# The read the code (and potentially modules) from /usr/share.
+files_read_usr_files(foreman_rails_t)
 
 # Allow Foreman to access log files (but not delete or truncate them). Ideally
 # only create and append should be allowed, however rake tasks also fall into
 # the same domain and those tasks often simply writes to log files.
-logging_search_logs(passenger_t)
-logging_list_logs(passenger_t)
-write_files_pattern(passenger_t, foreman_log_t, foreman_log_t)
-allow passenger_t foreman_log_t:dir { create_file_perms append_file_perms list_dir_perms search_dir_perms };
-allow passenger_t foreman_log_t:file { create_file_perms append_file_perms };
-
-# Allow Foreman to connect anywhere when bool is set
-tunable_policy(`passenger_can_connect_all',`
-    corenet_tcp_connect_all_ports(passenger_t)
+logging_search_logs(foreman_rails_t)
+logging_list_logs(foreman_rails_t)
+write_files_pattern(foreman_rails_t, foreman_log_t, foreman_log_t)
+allow foreman_rails_t foreman_log_t:dir { create_file_perms append_file_perms list_dir_perms search_dir_perms };
+allow foreman_rails_t foreman_log_t:file { create_file_perms append_file_perms };
+
+# Allow "logger" command for debugging
+kernel_dgram_send(foreman_rails_t)
+logging_create_devlog_dev(foreman_rails_t)
+
+# Systemd socket activation (Puma called with LISTEN_FDS environmental variable)
+allow foreman_rails_t self:netlink_route_socket { all_netlink_route_socket_perms };
+allow foreman_rails_t self:tcp_socket { all_tcp_socket_perms };
+allow foreman_rails_t self:udp_socket { all_udp_socket_perms };
+allow foreman_rails_t self:unix_dgram_socket { all_unix_dgram_socket_perms };
+corenet_tcp_bind_generic_node(foreman_rails_t)
+
+# Listening for HTTP communication on port 3000 (development default port)
```

Comment:
Also the one we currently use in production

Reply:
Changed the comment.